### ISSUE TITLE
using ${var} layout renderer in File target for `fileName` property creates file at wrong location

### ISSUE THREAD

**Issue by ggirard07** [NLog/NLog.Web#323]
**Type**:

- Bug

**NLog version**: 4.5.10

**NLog.Web/NLog.Web.AspNetCore version**: 4.7.0

**NLog.Extensions.Logging version**: From NLog.Web.AspNetCore (>= 1.3.0)

**Platform**: .NET Core 2 

**Current NLog config** (xml or C#, if relevant)

```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      autoReload="true"
      internalLogLevel="Off"
      internalLogFile="c:\temp\internal-nlog.txt">

  <variable name="logFolder" value="logs"/>

  <targets>
    <!--<target xsi:type="File" name="f" fileName="${logFolder}\nlog.log"-->
    <target xsi:type="File" name="f" fileName="${var:logFolder}\nlog.log"
            layout="${longdate}|${event-properties:item=EventId_Id}|${uppercase:${level}}|${logger}|${message} ${exception:format=tostring}" />
  </targets>

  <rules>
    <logger name="*" minlevel="Trace" writeTo="f" />
  </rules>
</nlog>
```

Sample to reproduce the issue available at https://github.com/ggirard07/NLogFilenameVarTest

**Steps to reproduce**
1. Set `NLogFilenameVarTest` as startup project
2. Start debug
3. Check file log location

**Expected result**
Log file should be at `ProjectRoot\bin\Debug\netcoreapp2.1\logs\nlog.log`

**Actual result**
Log file is located at `ProjectRoot\logs\nlog.log`

**More information**
Issue appear as soon as you use `${var:myVar}` renderer in `fileName` property.
If referencing the variable in `fileName` without the renderer as `fileName="${logFolder}\nlog.log"`, then everything works fine.

At first I though it was because using the layout renderer `${var:myVar}` wasn't supported in `fileName`, but then I noticed it was working without any issue in a console app running in .Net Core 2 too...
There is a console app project in the sample to demonstrate it is working properly in that case.

**Comment by 304NotModified** [comment]
what happens if you use 

`<variable name="logFolder" value="${basedir}\logs"/>`  and `${var:logFolder}`?

**Comment by ggirard07** [comment]
Seems like logs are back to the proper place at `ProjectRoot\bin\Debug\netcoreapp2.1\logs\nlog.log`.

Is it simply because the `${basedir}` is not padded properly in that case?
Not sure why the behavior is not the same between AspNetCore and a simple .Net Core console in that case

**Comment by 304NotModified** [comment]
I think the conversion from relative to absolute is done in differnent places and then the "currentdirectory" is different.

Thanks for testing, this is indeed a bug.

**Comment by snakefoot** [comment]
One should avoid modifying NLog variables at runtime. See also https://github.com/NLog/NLog/wiki/Var-Layout-Renderer

Instead use the Global Diagnostic Context (GDC) - https://github.com/NLog/NLog/wiki/Gdc-layout-renderer :

`GlobalDiagnosticsContext.Set("logFolder","someValue");`

And

`${gdc:item=logFolder}`

Regarding the difference in `AppDomain.BaseDirectory` (What `${basedir}` resolves to), then I guess it is the nature of NetCore. Try updating your csproj with the following for both projects:

```
  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <TargetLatestAspNetCoreRuntimePatch>true</TargetLatestAspNetCoreRuntimePatch>
    <RuntimeFrameworkVersion>2.1.3</RuntimeFrameworkVersion>
  </PropertyGroup>
```

**Comment by ggirard07** [comment]
@snakefoot my aim is to share most of my nlog configuration between several projects so I don't have to maintain in sync dozens of nlog.config every time I want to change some layout or target configurations.
To keep things flexible, I want projects to be able to override some configuration(s) defined in the common file. That's why I used the layout renderer as it seems to fit that need and was way more simple than the GDC thing.
I don't want to edit those variables at runtime through the NLog API. From my understanding, what I am trying to accomplish does not have the same implication as there is no threading/appcontext involved in the case I described.

**Comment by snakefoot** [comment]
I just saw that you used `${var:logFolder}` and I really recommend that you use `${logFolder}` or use `${gdc:item=logFolder}`. Just stay away from `${var:logFolder}`

Have you tested with enforcing NetCore SDK 2.1.3 ?

**Comment by ggirard07** [comment]
@snakefoot sorry for not testing sooner, I tried those in the ASP.Net Core project but resulted in no behavior changes.
Out of curiosity, what did you expect with this change? Even if it worked, I am not confident to update those kind of settings in my projects unless I am absolutely sure about the consequences (which I can't get in that case).

**Comment by 304NotModified** [comment]
polite bump @ggirard07

**Comment by ggirard07** [comment]
@304NotModified sorry for not testing proposed workaround but it seemed out of the original scope of the issue as it wasn't using `var:` syntax (`var_` instead).
I still gave it a try with the `var:` syntax instead, but with no luck.

I ended up using the `${basedir}\` workaround you provided me in [your first comment](https://github.com/NLog/NLog.Web/issues/323#issuecomment-429297216).
The original use case I reported was about using `var:`. If solution if to not use `var: ` syntax, then issue can simply be close as it is a "not supported"/"as designed"/"won't fix" case.

For my case I will live with the risk mentioned by @snakefoot for now. If something goes wrong, then I will simply revert back to duplicate my `nlog.config` files.
